script.js

```diff
@@ -4,403 +4,294 @@ of the properties make sense to filter on, and some to sort on.*/
-  {
-    name: 'Meat Stock',
-    cuisineType: 'american',
-    ingredients: [
-      '2.5 pounds beef marrow bones',
-      '1 large onion, quartered',
-      '2 carrots, sliced',
-      '1 leek, cleaned and sliced',
-      '2 celery stalks, sliced',
-      '2.5 pounds organic beef stew meat, cubed',
-      '2 tablespoons tomato paste',
-      '5 cloves garlic',
-      '2 bay leaves',
-      '3 sprigs thyme',
-      '3 sprigs Italian parsley',
-      '1/2 teaspoon black peppercorns'
-    ],
-    source: 'Food52',
-    totalTime: 60,
-    url: 'https://food52.com/recipes/3712-meat-stock',
-    image: './recipe-images/meat.jpg'
-  },
-  {
-    name: 'Homemade Meat Broth',
-    cuisineType: 'american',
-    ingredients: [
-      '1 teaspoon salt',
-      '1 carrot, peeled',
-      '1 medium onion, peeled',
-      '2 or 3 celery stalks',
-      '¼ red or yellow bell pepper, stripped of all its seeds',
-      '1 small boiling potato, peeled',
-      '1 ripe, fresh tomato, or 1 canned Italian plum tomato, drained of juice',
-      '5 pounds assorted pieces of meat and bones (see meat suggestions above), of which no less than 1½ pounds is all meat'
-    ],
-    source: 'Cookstr',
-    totalTime: 60,
-    url: 'http://www.cookstr.com/recipes/il-brodo-homemade-meat-broth',
-    image: './recipe-images/meat.jpg'
-  },
-  {
-    name: 'Spice-Rubbed Grilled Flap Meat (Sirloin Tip) Recipe',
-    cuisineType: 'south-american',
-    ingredients: [
-      '1 tablespoon whole black peppercorns, toasted',
-      '1 teaspoon coriander seed, toasted',
-      '1 teaspoon fennel seed, toasted',
-      '1 teaspoon cumin pods, toasted',
-      '1 teaspoon red pepper flakes',
-      '1/2 teaspoon dried oregano',
-      '2 medium cloves garlic, minced (about 2 teaspoons)',
-      '2 tablespoons vegetable or canola oil',
-      '1 whole flap meat steak, 2 to 2 1/2 pounds',
-      'Kosher salt'
-    ],
-    source: 'Serious Eats',
-    totalTime: 240,
-    url: 'http://www.seriouseats.com/recipes/2012/05/spice-rubbed-grilled-flap-meat-sirloin-tip-recipe.html',
-    image: './recipe-images/grilled.jpg'
+//Here we are listen for clicks on the buttons
+sortNewestBtn.addEventListener("click", (event) => {
+  //This one makes the hover effect stays on the button when sorting books
+  sortNewestBtn.classList.add("active") 
+  //This one removes the hover effect when another sorting-button is clicked
+  sortOldestBtn.classList.remove("active")
+  sortRatingBtn.classList.remove("active")
+  randomBtn.classList.remove("active")
+  //Here we sort books by year, newest to oldest. 
+  books.sort(function (a, b) {
+    return b.year - a.year;
+  });
+  booksDiv.innerHTML = "";
+  for (const book of books) {
+    booksDiv.innerHTML += renderBook(book);
+  }
+});
+
+sortOldestBtn.addEventListener("click", (event) => {
+  sortOldestBtn.classList.add("active") 
+  sortNewestBtn.classList.remove("active") 
+  sortRatingBtn.classList.remove("active")
+  randomBtn.classList.remove("active")
```

Comment:
Could these perhaps have been turned into a function that removes "active" for each sort button?